cedar-policy-symcc/src/symcc/term.rs

```diff
@@ -56,15 +59,15 @@ pub enum Term {
     Prim(TermPrim),
     Var(TermVar),
     None(TermType),
-    Some(Box<Term>),
+    Some(Arc<Term>),
     Set {
-        elts: BTreeSet<Term>,
+        elts: Arc<BTreeSet<Term>>,
```

Comment:
I'm not sure about this change. Why do we want `Arc` to be on the `BTreeSet`?

Reply:
Sorry I merged too fast. I can make suitable changes in the next PR.

Yeah I'm not completely sure either. I did this way since I see that, e.g., `ExprKind` in `cedar-policy-core` is doing something like that.
Are you concerned about the performance of this vs. just wrapping `Arc<Term>`, or something else?

Reply:
One possible answer is that cloning `Arc<BTreeSet<_>>` is much faster than cloning `BTreeSet<Arc<_>>`.  If we clone this a lot, this could be a substantial part of the performance win described in this PR.

Reply:
Great, let's leave it as is then. Thanks for testing it!